buildtools/cli.go

```diff
@@ -622,6 +644,88 @@ func CreateBuildConfigurationWithModule(c *cli.Context) (buildConfigConfiguratio
 	return
 }
 
+func dockerCmd(c *cli.Context) error {
+	args := cliutils.ExtractCommand(c)
+	var cmd, image string
+	// We may have prior flags before push/pull commands for the docker client.
+	for _, arg := range args {
+		if !strings.HasPrefix(arg, "-") {
+			if cmd == "" {
+				cmd = arg
+			} else {
+				image = arg
+				break
+			}
+		}
+	}
+	switch cmd {
+	case "pull":
```

Comment:
I suggest restructuring the command similar to #1416, i.e;
A `docker` name space that has `pull`, `push` and `scan` as subcommands. This way the flag parsing is done by gangsta, and an appropriate help can be shown for each subcommand.
(I did not review this PR)

Reply:
As a result of a bug in `urfave/cli v1.22.5` whereby flags are getting parsed depside the property `SkipFlagParsing: true,` we cannot implement this now.